### DISCUSSION TITLE
Need help with Scenario 5: MLflow Tracking Server enabled with proxied artifact storage access

### DISCUSSION THREAD

**Discussion by bhavik161** [mlflow/mlflow#8659] [Q&A]
Hi,
I have deployed MLflow Tracking server on AWS ECS.  The data base is hosted in RDS Postgress.  I start the MLFlow Tracking server using the command:
`
mlflow server --backend-store-uri postgresql://{user_name}:{password}@{host}:{port}/{dbname} " \
        " --default-artifact-root s3://{bucket}/app/mlflowdefault/ --serve-artifacts --host 0.0.0.0 --port 8000"
`

the ECS has permission to access the S3.
the Client running MLFlow code does not have permission to S3.
When I try to save the model I get access denied and it seems that instead of using proxy to use tracking server permission to access S3 it is trying to access s3 using client credential.  

`
import os
from sklearn.datasets import make_regression
from sklearn.ensemble import RandomForestRegressor
from sklearn.metrics import mean_squared_error
from sklearn.model_selection import train_test_split

import mlflow
import mlflow.sklearn
from mlflow.models.signature import infer_signature
#os.environ['MLFLOW_ALLOW_FILE_URI_AS_MODEL_VERSION_SOURCE']='True'
os.environ['MLFLOW_TRACKING_INSECURE_TLS'] = 'true'

os.environ['MLFLOW_S3_UPLOAD_EXTRA_ARGS']='{"ServerSideEncryption": "aws:kms", "SSEKMSKeyId": "alias/abc/app/xyz"}'
mlflow.set_tracking_uri("https://tracking-server.com")

experiment_name="exp-16"
experiment=mlflow.set_experiment(experiment_name)

with mlflow.start_run( run_name="RUN1",
            experiment_id=experiment.experiment_id, nested=True
                     ) as run:
    X, y = make_regression(n_features=4, n_informative=2, random_state=0, shuffle=False)
    X_train, X_test, y_train, y_test = train_test_split(
        X, y, test_size=0.2, random_state=42
    )
    params = {"max_depth": 2, "random_state": 42}
    model = RandomForestRegressor(**params)
    model.fit(X_train, y_train)

    # Infer the model signature
    y_pred = model.predict(X_test)
    signature = infer_signature(X_test, y_pred)

    # Log parameters and metrics using the MLflow APIs
    mlflow.log_params(params)
    mlflow.log_metrics({"mse": mean_squared_error(y_test, y_pred)})
    
    mlflow.sklearn.log_model(
        sk_model=model,
        #local_dir='abc',
        artifact_path="https://tracking-server.com/mlartifactse",
        signature=signature,
        registered_model_name=experiment_name,
    )
`
I am getting the following error:

S3UploadFailedError: Failed to upload /tmp/tmpea4bxeng/model/model.pkl to bucket/app/mlflowdefault/27/83194c01e9d9402db5d28f11c8002f24/artifacts/https://tracking-server.com/mlartifactse/model.pkl: An error occurred (AccessDenied) when calling the PutObject operation: Access Denied

It seems that its not passing the model to tracking server and trying to save from client only

**Comment by NWTRP** [comment]
I'm running into the same issue and using the exact same combination of arguments as here:
https://mlflow.org/docs/latest/tracking.html#scenario-5-mlflow-tracking-server-enabled-with-proxied-artifact-storage-access

I am either missing something or this documentation is incorrect?

**Comment by kavita1205** [comment]
@bhavik161 and @NWTRP did you guys get any solution for this issue. I am also facing the same issue.

**Comment by rzalawad** [comment]
I believe you need to set --artifacts-destination to the s3 path. --default-artifact-root is for when you are not proxying but you still want to have those paths show up in mlflow UI
